lib/src/Uncertainty/Distribution/GeneralizedExtremeValueFactory.cxx

```diff
@@ -82,6 +94,899 @@ GeneralizedExtremeValue GeneralizedExtremeValueFactory::buildAsGeneralizedExtrem
+  const Scalar kst = (2.0 * bpwm1 - mean) / (3.0 * bpwm2 - mean) - std::log(2.0) / std::log(3.0);
+  const Scalar xi0 = -(7.859 + 2.9554 * kst) * kst;
+  const Point x0({xi0});
+
+  const GeneralizedExtremeValueProfileLikelihoodEvaluation profileLikelihoodEvaluation(sample, mean, bpwm1);
+  const Function objective(profileLikelihoodEvaluation.clone());
+  OptimizationProblem problem(objective);
+  problem.setMinimization(false);
+
+  // solve optimization problem
+  Cobyla solver(problem);
+  solver.setProblem(problem);
+  solver.setMaximumEvaluationNumber(ResourceMap::GetAsUnsignedInteger("GeneralizedExtremeValueFactory-MaximumEvaluationNumber"));
+  solver.setStartingPoint(x0);
+  solver.run();
+
+  // rerun once to get optimal (mu, sigma) at optimal xi
+  const Scalar xi = solver.getResult().getOptimalPoint()[0];
+  profileLikelihoodEvaluation(solver.getResult().getOptimalPoint());
+  Point optimalParameter(profileLikelihoodEvaluation.getOptimalPoint());
+  optimalParameter.add(xi);
+
+  const Distribution distribution(buildAsGeneralizedExtremeValue(optimalParameter));  
+  Distribution parameterDistribution(MaximumLikelihoodFactory::BuildGaussianEstimator(distribution, sample));
+  parameterDistribution.setDescription({"mu", "sigma", "xi"});
+  const Scalar logLikelihood = solver.getResult().getOptimalValue()[0];
+  ProfileLikelihoodResult result(distribution, parameterDistribution, logLikelihood, objective, xi);
+  return result;
+}
+
+GeneralizedExtremeValue GeneralizedExtremeValueFactory::buildMethodOfProfileLikelihoodMaximization(const Sample & sample) const
+{
+  const Distribution distribution(buildMethodOfProfileLikelihoodMaximizationEstimator(sample).getDistribution());
+  return buildAsGeneralizedExtremeValue(distribution.getParameter());
+}
+
+
+
+class GeneralizedExtremeValueRMaximaLikelihoodEvaluation : public EvaluationImplementation
+{
+public:
+  GeneralizedExtremeValueRMaximaLikelihoodEvaluation(const Sample & sample,
+                                                     const UnsignedInteger r)
+  : EvaluationImplementation()
+  , sample_(sample)
+  , R_(sample.getDimension())
+  , r_(r)
+  , m_(sample.getSize())
+  {
+    if ((r < 1) || (r > R_))
+      throw InvalidArgumentException(HERE) << "1<=r<=R";
+  }
+
+  GeneralizedExtremeValueRMaximaLikelihoodEvaluation * clone() const override
+  {
+    return new GeneralizedExtremeValueRMaximaLikelihoodEvaluation(*this);
+  }
+
+  UnsignedInteger getInputDimension() const override
+  {
+    return 3;
+  }
+
+  UnsignedInteger getOutputDimension() const override
+  {
+    return 1;
+  }
+
+  Point operator() (const Point & parameter) const override
+  {
+    const Scalar mu = parameter[0];
+    const Scalar sigma = parameter[1];
+    const Scalar xi = parameter[2];
+
+    // in practice it never goes here
+    if (std::abs(xi) < SpecFunc::Precision)
+      throw NotYetImplementedException(HERE) << "Gumbel case";
```

Comment:
What is the problem with the Gumbel case?

Reply:
no problem